worker-workflow/src/main/java/com/github/cafdataprocessing/WorkflowEvaluator.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2015-2017 EntIT Software LLC, a Micro Focus company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.github.cafdataprocessing;
+
+import com.hpe.caf.worker.document.DocumentWorkerConstants;
+import com.hpe.caf.worker.document.JavaScriptDocumentPostProcessor;
+import com.hpe.caf.worker.document.exceptions.PostProcessingFailedException;
+import com.hpe.caf.worker.document.model.Document;
+import com.hpe.caf.worker.document.model.ResponseOptions;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Evaluates a JavaScript workflow against a document, updating document based on matched actions in the workflow
+ */
+public class WorkflowEvaluator
+{
+    /**
+     * Evaluates the provided document against the JavaScript workflow and records the workflow storage reference on
+     * the document for execution in post processing.
+     * @param document The document to evaluate against the workflow. May be modified by evaluation
+     *                 e.g. fields indicating action being executed, the queue to send document to, the post processing
+     *                 script to execute.
+     * @param workflowAsJavaScript JavaScript representation of a workflow.
+     * @param workflowStorageRef storage reference for the workflow passed. Will be recorded as a post processing script
+     *                           if none is already set.
+     * @throws PostProcessingFailedException if there is a failure in workflow script execution.
+     */
+    public static void evaluate(Document document, String workflowAsJavaScript, String workflowStorageRef)
+            throws PostProcessingFailedException
+    {
+        final JavaScriptDocumentPostProcessor jsPostProcessor = new JavaScriptDocumentPostProcessor(workflowAsJavaScript);
+        jsPostProcessor.postProcessDocument(document);
+        setWorkflowPostProcessingScript(document, workflowStorageRef);
```

Comment:
We don't need to do this right now but raise a Jira to add some functionality to the Document Worker Framework so that you don't need this dependency on `JavaScriptDocumentPostProcessor`.  Really the worker should only need to depend on `worker-document-interface`, but obviously you are having to take a dependency on `worker-document` itself, which is meant to be an internal module, in order to use this class.

Reply:
Raised, https://jira.autonomy.com/browse/CAF-3963.